SignalMessaging/profiles/OWSProfileManager.m

```diff
@@ -535,13 +535,16 @@ - (void)updateServiceWithProfileName:(nullable NSString *)localProfileName
     OWSAssertDebug(successBlock);
     OWSAssertDebug(failureBlock);
 
-    [[LKGroupChatAPI setDisplayName:localProfileName on:LKGroupChatAPI.publicChatServer]
-    .thenOn(dispatch_get_main_queue(), ^() {
-        successBlock();
-    })
-    .catchOn(dispatch_get_main_queue(), ^(NSError *error) {
-        failureBlock(error);
-    }) retainUntilComplete];
+    __block NSDictionary *chats;
+    [SSKEnvironment.shared.primaryStorage.dbReadConnection readWithBlock:^(YapDatabaseReadTransaction * _Nonnull transaction) {
+        chats = [LKDatabaseUtilities getAllGroupChats:transaction];
+    }];
+    
+    for (LKGroupChat *chat in chats.allValues) {
```

Comment:
Just realised something, we should probably reduce the chats down to a set of servers since there could be a case where you can have 2 chats with the same server (E.g when they both have different channels)